libkbfs/folder_branch_ops.go

```diff
@@ -592,15 +659,82 @@ func (fbo *folderBranchOps) setHeadLocked(ctx context.Context,
 			fbo.identifyDone = false
 		}()
 	}
-	if !wasReadable && md.IsReadable() {
-		// Let any listeners know that this folder is now readable,
-		// which may indicate that a rekey successfully took place.
-		fbo.config.Reporter().Notify(ctx, mdReadSuccessNotification(
-			md.GetTlfHandle().GetCanonicalName(), md.ID.IsPublic()))
+
+	return nil
+}
+
+// setHeadPredecessorLocked is for when we're unstaging updates.
+func (fbo *folderBranchOps) setHeadPredecessorLocked(ctx context.Context,
+	lState *lockState, md *RootMetadata) error {
+	fbo.mdWriterLock.AssertLocked(lState)
+	fbo.headLock.AssertLocked(lState)
+	if fbo.head == nil {
+		return errors.New("Unexpected nil head in setHeadPredecessorLocked")
+	}
+	if fbo.head.Revision <= MetadataRevisionInitial {
+		return fmt.Errorf("setHeadPredecessorLocked unexpectedly called with revision %d", fbo.head.Revision)
+	}
+	err := md.CheckValidSuccessor(fbo.config, fbo.head)
+	if err != nil {
+		return err
+	}
+
+	oldHandle := fbo.head.GetTlfHandle()
+	newHandle := md.GetTlfHandle()
+
+	// The new, older handle might become less resolved.
```

Comment:
"new, older" is tough to parse.  Maybe drop "new"?  (And also maybe "become" -> "be" for less confusion.)